### DISCUSSION TITLE
Prisma Instrospect/Generate every time after DB Migration ?

### DISCUSSION THREAD

**Discussion by juanitoddd** [prisma/prisma#6621] [Q&A] [answered]
Hello dear Devs,

 I just started to use Prisma to query the Database from several microservices in my architecture.

DB: MySQL

I already use a Backend tool to manage the Database migrations, and I notice that, after running my migrations, the Prisma client stoped to work throwing _Attempted to serialize scalar 'null' with incompatible type 'String'_ kind of messages... hmm weird.

After reading and trying things out, I finally ran _prisma introspect_ and _prisma generate_ and things went back to normal

My question to you, is: 

Every time there is a change in DB structure, these commands have to be run for Prisma to work properly ?

Have a nice day

**Comment by janpio** [comment, marked as answer]
Yes, the Prisma schema needs to be updated with the _real_ database schema, and the Prisma Client needs to be regenerated afterwards so it knows how to talk to the database. 

(There might be cases where things keeps working - but that is just accidental as having to little, too many or wrong fields or tables is not causing the database queries to crash. But the information use for that is definitely out of date.)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
